### ISSUE TITLE
user specified svg directory?

### ISSUE THREAD

**Issue by mohkale** [sylvainmetayer/jekyll-fontawesome-svg#13]
It'd be nice to be able to use this plugin to include svgs from a custom directory.

so... you'd have the following in your _config.yml

```yaml
fontawesome:
  userdir: assets/svgs/
```

and then you could include the svg at `assets/svgs/foo.svg` by writing `{% fa_svg fac.foo %}` (c for custom) or you could use a namespace like syntax to expand into subdirectories. so `assets/svgs/foo/bar.svg` would be `{% fa_svg fac.foo.bar %}`.

thoughts?

**Comment by sylvainmetayer** [comment]
Hello,

Thanks for your interest in this project ! Sadly, I think I don't fully understand what you're trying to achieve.

This gem does not include svg files into your generated site. It only render used icons (with `<symbol>` tag) at the bottom of the page the used icons for the current page (with `{% fa_svg_generate %}`. 

I don't understand why you would then change the location of the assets svg, since their are never used on your site.

Could you please details your need ?

**Comment by mohkale** [comment]
the end goal would be using jekyll-fontawesome-svg to include your **own** svgs into the page, at the moment we're limited to the ones offered by this plugin. Say there's an SVG I want on my site but you don't have it in this repository. I can either switch to [jekyll-inline-svg](https://github.com/sdumetz/jekyll-inline-svg) and add the SVG to some path in my src directory, or I can avoid the extra dependency and just tell jekyll-fontawesome-svg that I want you to also look for SVGs to include in this directory alongside the ones you're already using.

**Comment by sylvainmetayer** [comment]
Ok, I get it. However, I think it is not the responsibility of this plugin to handle arbitrary svg, others than those of fontawesome. I already created another gem to handle the [academicons](https://github.com/sylvainmetayer/jekyll-academicons-svg) icons to avoid name confusion about the goal of each gem.

However, you raised an interesting use case which would be to create a mix of these 2 gems, where you can render any svg, but where a svg displayed multiple time on the same page would be rendered at the bottom of a page only once (which from what I've seen does not seem to be the case with the jekyll-inline-svg ?) with a `generate` tag and use symbol to make a reference to the icons.

What do you think ?

**Comment by mohkale** [comment]
I haven't really needed academicons, so I'm not sure. It would be cool to only need to define SVGs once, but I'd still really like the ability to use my own SVGs. I agree custom SVG locations does seem to be out of the purview of this plugin. but it's a small change. All you really need to do is register a new tag and reuse the same functionality for finding SVGs in the assets directory as you would in the users custom directory.

**Comment by sylvainmetayer** [comment]
If this functionality is out of scope, I can't add it as users who rely on this (this plugin was first designed to answer an issue at [this Jekyll template](https://github.com/sergiokopplin/indigo/)) may encountered issues due to this. 

As I told in my previous comment, one solution would be to create another gem (in a separated repository) to answer this specific need, and create a plugin which could answer your need. If you are a developer, this is a matter of a few days to create it, or maybe less if you are more familiar than I am with Ruby and Jekyll environment.

**Comment by mohkale** [comment]
understood. I'll get onto it soon. 😄.

**Comment by sylvainmetayer** [comment]
If you could please keep me posted when this will be ready, I'd appreciate that and would be curious to look into it ! :smile:

**Comment by mohkale** [comment]
this didn't really seemed so big that it merits a whole plugin. I've monkey-patched the features I want into the `FontAwesomeIcon` class. For anyone who wants similair functionality, you can copy the following into your _plugins directory.

```ruby
# frozen_string_literal: true

require 'yaml'

module Mohkale
  module Plugins
    # make jekyll-fontawesome-svg have the option to include custom SVGs from
    # users SVG directory.
    module FontAwesomeSvgCustomDirectory
      def to_filename
        return super unless to_font_awesome_type == 'fac'

        File.join(custom_svg_directory, to_icon_name + ".svg")
      end

      def self.custom_svg_directory
        unless @custom_svg_directory
          dir = fa_svg_config["custom_dir"] or raise "no custom SVG directory configured"
          @custom_svg_directory = File.join(src_directory, dir)
        end

        @custom_svg_directory
      end

      def custom_svg_directory
        FontAwesomeSvgCustomDirectory.custom_svg_directory
      end

      private

      def self.config_path
        @config_path ||= File.join(Dir.pwd, "_config.yml")
      end

      def self.config
        @config ||= File.open(config_path, "r") { |file| YAML.load(file.read) }
      end

      def self.src_directory
        @src_directory ||= config["source"] || Dir.pwd
      end

      def self.fa_svg_config
        @fa_svg_config ||= config["fa_svg"] || {}
      end

      # make class methods accessible through instance.

      def config
        FontAwesomeSvgCustomDirectory.config
      end

      def fa_svg_config
        FontAwesomeSvgCustomDirectory.fa_svg_config
      end

      def src_directory
        FontAwesomeSvgCustomDirectory.src_directory
      end
    end

    begin
      Jekyll::FontAwesome::Svg::FontAwesomeIcon.prepend(FontAwesomeSvgCustomDirectory)
    rescue NameError
    end
  end
end
```

I've had to reparse the users config file because this plugin doesn't pass the context to the SVG instance, but aside from that it's quite straight forward. You specify a custom directory in you _config.yml file, which is relative to the sites src directory.

I've set:

```yaml
fa_svg:
  custom_dir: assets/fa-svg/

# also probably want to exclude fa icons.
exclude:
  - assets/fa-svg/
```
 and then on your posts you can write.

```markdown
---
---
# Hello World
{% fa_svg fac.fa-pacman %}

{% fa_svg_generate %}
```

and it'll load the SVG at `assets/fa-svg/pacman.svg` into the page 😄.
